Drupal 11 Support

svicervlad opened the pull request:
## Summary

This PR adds comprehensive Drupal 11 compatibility to the Activity Finder module and all its sub-modules. Includes deprecation fixes, proper dependency injection, and support for Drupal 8 through 11.

## Changes

### 1. Deprecation Fixes (Issue #2932520)
- Replaced deprecated `watchdog_exception()` function with modern error logging approach
- Injected `LoggerChannelInterface` for proper dependency injection
- Using `Error::logException()` with injected logger for Drupal 10.1+ compatibility

### 2. Module Version Support
Updated `core_version_requirement` to `^8 || ^9 || ^10 || ^11` for all modules:
- Activity Finder (openy_activity_finder)
- OpenY Paragraph Activity Finder (openy_prgf_activity_finder)
- OpenY Paragraph Activity Finder v4 (openy_prgf_activity_finder_4)
- OpenY Paragraph Activity Finder v4 Demo (openy_prgf_af4_demo)
- OpenY Paragraph Camp Finder (openy_prgf_camp_finder)
- OpenY Paragraph Activity Finder Search (openy_paragraph_activity_finder_search)

Note: lb_activity_finder already supports Drupal 11 (^10 || ^11).

### 3. ConfigFormBase Constructor Fix
- Properly injected `TypedConfigManagerInterface` to `ConfigFormBase` constructor
- Fixed TypeError that occurred when accessing the Activity Finder settings form
- All dependencies follow Drupal best practices

## Commits

1. **fix: replace deprecated watchdog_exception with backwards-compatible error logging** (2767cbb)
   - Initial deprecation fix with backwards compatibility wrapper

2. **feat: add Drupal 11 support to activity finder module and sub-modules** (90605c2)
   - Updated all info.yml files with Drupal 11 support

3. **refactor: replace watchdog_exception with injected logger service** (aa8b1f2)
   - Refactored to use modern dependency injection approach
   - Removed unnecessary DeprecationHelper wrapper

4. **fix: inject TypedConfigManagerInterface to ConfigFormBase constructor** (f7c9bbe)
   - Fixed constructor signature for proper ConfigFormBase compatibility
   - Resolved TypeError in settings form

## Test Plan

- [x] Code reviewed for Drupal deprecations
- [x] Tested on SCCY site with Drupal 11.1.8
- [x] Activity Finder module enables without errors
- [x] Settings form accessible at `/admin/openy/settings/activity-finder`
- [x] Logger injection working properly
- [x] All sub-modules compatible with Drupal 11

## Related Issues

- Fixes Drupal deprecation #2932520 (watchdog_exception deprecation)
- Prepares Activity Finder for full Drupal 11 compatibility

svicervlad committed:
fix: replace deprecated watchdog_exception with backwards-compatible error logging

Refactor watchdog_exception() to use DeprecationHelper::backwardsCompatibleCall()
with Error::logException() for Drupal 10.1.0+ compatibility while maintaining
backwards compatibility with older versions. Prepared for Drupal 11 upgrade.

svicervlad committed:
feat: add Drupal 11 support to activity finder module and sub-modules

Update core_version_requirement to ^8 || ^9 || ^10 || ^11 for:
- Activity Finder (openy_activity_finder)
- OpenY Paragraph Activity Finder (openy_prgf_activity_finder)
- OpenY Paragraph Activity Finder v4 (openy_prgf_activity_finder_4)
- OpenY Paragraph Activity Finder v4 Demo (openy_prgf_af4_demo)
- OpenY Paragraph Camp Finder (openy_prgf_camp_finder)
- OpenY Paragraph Activity Finder Search (openy_paragraph_activity_finder_search)

Note: lb_activity_finder already supports Drupal 11 (^10 || ^11).

svicervlad committed:
refactor: replace watchdog_exception with injected logger service

Fix Drupal deprecation issue #2932520 by:
- Injecting LoggerChannelInterface into SettingsForm
- Using Error::logException() with the injected logger
- Removing deprecated watchdog_exception() call
- Removing unnecessary DeprecationHelper wrapper

This modernizes the codebase to follow Drupal 10.1+ best practices and
prepares for full Drupal 11 compatibility.

svicervlad committed:
fix: inject TypedConfigManagerInterface to ConfigFormBase constructor

Fix TypeError where ConfigFormBase requires both ConfigFactoryInterface and
TypedConfigManagerInterface in the constructor. Added proper dependency
injection of typed config manager service.

This resolves the error when accessing the Activity Finder settings form.

podarok commented on modules/openy_paragraph_activity_finder_search/openy_paragraph_activity_finder_search.info.yml:
```diff
@@ -2,7 +2,7 @@ name: 'OpenY Paragraph Activity Finder Search'
 type: module
 package: "YMCA Website Services"
-core_version_requirement: ^8 || ^9 || ^10
+core_version_requirement: ^8 || ^9 || ^10 || ^11
```
```suggestion
core_version_requirement: ^10 || ^11
```

and do this in all info.yml files accross the codebase, please

svicervlad committed:
refactor: update core_version_requirement to ^10 || ^11 across all modules

Remove support for Drupal 8 and 9 as per code review feedback. Update all
info.yml files to support only Drupal 10 and 11, aligning with the modern
maintenance and testing requirements.
